processing/src/main/java/io/druid/query/timeseries/TimeseriesQueryQueryToolChest.java

```diff
@@ -342,6 +353,10 @@ public boolean isCacheable(TimeseriesQuery query, boolean willMergeRunners)
             timeseriesQuery = timeseriesQuery.withDimFilter(timeseriesQuery.getDimensionsFilter().optimize());
             queryPlus = queryPlus.withQuery(timeseriesQuery);
           }
+          if (timeseriesQuery.getLimit() < Integer.MAX_VALUE) {
```

Comment:
I don't think this block of code is really doing anything? It looks like it is re-creating the timeseriesQuery with the same limit that it already has.

Reply:
removed